drivers/spi/Kconfig

```diff
@@ -112,6 +112,8 @@ source "drivers/spi/Kconfig.xec_qmspi"
 
 source "drivers/spi/Kconfig.gecko"
```

Comment:
Maybe we could rename `Kconfig.gecko` in `Kconfig.gecko_usart`. It would be great to also rename `CONFIG_SPI_GECKO`.

BTW, `gecko` does not really make sense. We try now to rather use `silabs`. @asmellby @jhedberg do you agree?

Reply:
Done, I left it as gecko because that's how it is in the other configurations and drivers, but if necessary I can modify it.

Thanks for your feedback!